src/engraving/libmscore/select.cpp

```diff
@@ -698,8 +698,44 @@ void Selection::setRangeTicks(const Fraction& tick1, const Fraction& tick2, staf
 
 void Selection::update()
 {
-    for (EngravingItem* e : _el) {
-        e->setSelected(true);
+    if (element()) {
+        element()->setSelected(true);
+        updateState();
+        return;
+    }
+
+    if (_activeSegment && _endSegment && !_el.empty() && _activeSegment->tick() < _endSegment->tick()) {
+        std::vector<std::vector<EngravingItem*> > el;
+        int i = 0;
+        while (i < static_cast<int>(_el.size())) {
+            std::vector<EngravingItem*> elements = { _el[i] };
+
+            if (i != static_cast<int>(_el.size()) - 1) {
+                int j = i + 1;
+                while (j < static_cast<int>(_el.size()) && _el[i]->tick() == _el[j]->tick()) {
+                    elements.emplace_back(_el[j]);
+                    j++;
+                }
+                if (j < static_cast<int>(_el.size()) && _el[i]->tick() != _el[j]->tick()) {
+                    i = j;
+                } else {
+                    i = j + 1;
+                }
+            } else {
+                i++;
+            }
+            el.emplace_back(elements);
+        }
+
+        for (auto it = el.rbegin(); it != el.rend(); it++) {
+            for (EngravingItem* e : *it) {
+                e->setSelected(true);
+            }
+        }
```

Comment:
We really need some comments in this function to explain what you are doing and why it is necessary.

1. It's really hard to see what you are doing differently to before. If it wasn't for the `r` in `rbegin()` and `rend()` then your 38 lines of new code would be exactly equivalent to the 2 lines that were already there.

2. Even once I spotted the `r` and realised that you were calling `e->setSelected(true)` on elements in a slightly different order, it's not obvious why the order matters.

You're also relying on the `_el` list being in a certain initial ordering. It works most of the time, but it breaks when there's an articulation on the final note in the range selection.

I think there is a much simpler way to solve it that doesn't rely on `_el` being ordered in a certain way:
```C++
void Selection::update()
{
    for (EngravingItem* e : _el) {
        e->setSelected(true); // also tells accessibility that e has focus
    }
    // Only one element can have focus at a time, so currently the final
    // element in _el has focus. That's ok for a LIST selection because it
    // corresponds to the last element the user clicked on.
    if (ChordRest* cr = activeCR()) {
        // User is performing a RANGE selection. Let's focus a note/rest in the activeCR.
        EngravingItem* e = cr->isRest() ? toEngravingItem(cr) : toChord(cr)->upNote();
        assert(e->selected()); // was selected in loop above (e is somewhere in _el)
        e->setSelected(true); // HACK: select it again so accessibility thinks it has focus
    }
    updateState();
}
```